lmdeploy/pytorch_poc/models/patch.py

```diff
@@ -359,21 +379,28 @@ def patch(
     extra_args_str = ' '.join(f'{arg}=None,' for arg in extra_args)
     context_update_str = ' '.join(f'{arg}={arg},' for arg in extra_args)
 
-    wrap_forward_src = f"""\
+    wrap_forward_src = f"""
 from functools import wraps
-old_forward = model.forward
+# old_forward = model.forward
+old_forward = type(model).forward
 @wraps(old_forward)
-def wrap_forward(*args, {extra_args_str} **kwargs):
+def wrap_forward(self, *args, {extra_args_str} **kwargs):
     global _patch_context
     _patch_context.update({context_update_str})
 
-    output = old_forward(*args, **kwargs)
+    output = old_forward(self, *args, **kwargs)
 
     _patch_context.clear()
 
     return output
-model.forward = wrap_forward
-    """
+# model.forward = wrap_forward
+
+attrs = dict(type(model).__dict__)
```

Comment:
Update the dict of class might leads to unexpected behavior if user want both origin class and patched class.

Reply:
I tested the new patch with llama, internlm, chatglm2 and baichuan. All works.